wire-ios-data-model/Source/ManagedObjectContext/Migration/2.110-2.111/CoreDataAction2111.swift

```diff
@@ -0,0 +1,47 @@
+////
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+
+import Foundation
+
+protocol CoreDataAction2111 {
+    func primaryKey(for object: NSManagedObject, entityName: String) -> String
+}
+
+extension CoreDataAction2111 {
+
+    func primaryKey(for object: NSManagedObject, entityName: String) -> String {
+
+        let remoteIdentifierData = object.value(forKey: ZMManagedObject.remoteIdentifierDataKey()) as? Data
+
+        switch entityName {
+        case ZMUser.entityName(), ZMConversation.entityName():
+            let path = entityName == ZMUser.entityName() ? #keyPath(ZMUser.domain) : #keyPath(ZMConversation.domain)
+
+            let domain = object.value(forKeyPath: path) as? String
+            return ZMManagedObject.primaryKey(from: remoteIdentifierData.flatMap(UUID.init(data: )), domain: domain)
```

Comment:
opinion: it's better understandable and easier to maintain to use the cases individually and concrete key paths and funcs:
```suggestion
        case ZMUser.entityName():
            let domain = object.value(forKeyPath: #keyPath(ZMUser.domain)) as? String
            let remoteIdentifier = remoteIdentifierData.flatMap(UUID.init(data:))
            return ZMUser.primaryKey(from: remoteIdentifier, domain: domain)

        case ZMConversation.entityName():
            let domain = object.value(forKeyPath: #keyPath(ZMConversation.domain)) as? String
            let remoteIdentifier = remoteIdentifierData.flatMap(UUID.init(data:))
            return ZMConversation.primaryKey(from: remoteIdentifier, domain: domain)
```